src/CommitmentsV2/SFA.DAS.CommitmentsV2/Extensions/QueryableApprenticeshipsExtensions.cs

```diff
@@ -178,8 +183,105 @@ private static IQueryable<Apprenticeship> WithAlertsEmployer(this IQueryable<App
                 return apprenticeships.Where(app => app.Cohort.ProviderId == identifier.ProviderId);
             }
 
-            return identifier.EmployerAccountId.HasValue ? 
+            return identifier.EmployerAccountId.HasValue ?
                 apprenticeships.Where(app => app.Cohort.EmployerAccountId == identifier.EmployerAccountId) : apprenticeships;
         }
+
+        public static IQueryable<Apprenticeship> FilterApprenticeshipByAlert(IQueryable<Apprenticeship> apprenticeships, Alerts alert, bool isProvider)
+        {
+            switch (alert)
+            {
+                case Alerts.IlrDataMismatch:
+                    return apprenticeships.Where(a =>
+                        (
+                            a.DataLockStatus.Any(x =>
```

Comment:
When this did use the extensions it was little bit self documenting with the extension method names - now it's not at all self documenting - I think this reads a Course data lock or price only data lock which is currently not triaged by the provider - but I only know that because I've just worked on the data lock stuff - mostly people would not have a clue what this does - I think it need comments adding to explain the logic

Reply:
Comment added